### DISCUSSION TITLE
Transcript table

### DISCUSSION THREAD

**Discussion by skiniry** [skiniry/Trips-Viz#2] [Ideas]
When users enter a gene name on the single transcript or comparison plot page they are presented with a table of transcript id's and basic information about each one, this information could be more detailed to show users which transcript is best supported by the data.

**Comment by JackCurragh** [comment]
The prospect of calculating TPMs /OPMs for each transcript of each gene of each organism on Trips would likely be the most informative option but calculations are computationally expensive (at least with my current scripts). We could parallelise it and have it run on Poitin in the background but I have no idea how long that will take. 

What did you think of any data fit idea from yesterdays seminar? I can reexplain it here if you would like. It should be less computationally heavy but will still require reading in each read file one by one. I will have to implement it before we would have any idea how long that would take.

**Reply by solenoozerec** [reply]
There are two issues in selection (1) visualisation and (2) data fit. They don't have to be solved all at once.  ​You were showing a JBrowse applet, embedding it to trips would already be of great help as it will provide better guidance in comparison with UTRs/CDS lengths. APRIS isoform could be differentially highlighted for example.

**Reply by skiniry** [reply]
> What did you think of any data fit idea from yesterdays seminar? I can reexplain it here if you would like. It should be less computationally heavy but will still require reading in each read file one by one. I will have to implement it before we would have any idea how long that would take.

Definitely worth a try but I'd wonder if it would be any quicker really, Is the bulk of the time taken for TPM/OPM tables is taken up by just reading in files or is it more in the calculations? BTW, the TPM/OPM tables are working on the live version of trips now, they were broken there for a while. It's accessible under the test user only. Seems to be working fine for some genes but for others it throws an error, IFRD1 and UCP2 for example give the following error:

File "/home/DATA/www/tripsviz/tripsviz/tripsSplice.py", line 221, in genomic_orf_coordinate_ranges
    orf_structures[transcript] = get_orf_exon_structure(start_stop, transcript_coordinates)
File "/home/DATA/www/tripsviz/tripsviz/tripsSplice.py", line 113, in get_orf_exon_structure
    start, stop = start_stop
TypeError: 'NoneType' object is not iterable


Do you have this running on your local version, do you get the same error for those genes?

**Reply by JackCurragh** [reply]
I haven't worked on my local version of trips in a while since its on my old laptop. I should be able to clone it from my github and see how it goes. 

Those scripts leave a lot to be desired so I might just rewrite them if they are giving issues.

**Reply by JackCurragh** [reply]
Okay this has been corrected. Ill update the files on poitin too. 

Working on a script to automate the calculations on OPM and TPM values within all sqlitedicts. When I have something worth considering I will make a repo on my account so you can have a look
